### PR TITLE
engine: move setFrontFaceWindingInverted from View to MaterialInstance

### PR DESCRIPTION
**Pull request by poweifeng**
Moving setFrontFaceWindingInverted to MaterialInstance will enable finer control over face inversion and aligns better with Vulkan's pipeline definition (see VkGraphicsPipelineCreateInfo).

### CODE REVIEW

File: filament/src/RenderPass.cpp [hunk 1]
```diff
@@ -579,10 +576,12 @@ RenderPass::Command* RenderPass::generateCommandsImpl(uint32_t extraFlags,
             auto const& morphTargets = morphing.targets[pi];
             FMaterialInstance const* const mi = primitive.getMaterialInstance();
             FMaterial const* const ma = mi->getMaterial();
+            bool const invertedFrontFaces =
+                    mi->isFrontFaceWindingInverted() ^ soaVisibility[i].reversedWindingOrder;
```
Comment (pixelflinger) [inline filament/src/RenderPass.cpp:580]: `soaVisibility[i]` can be hoisted outside the loop.

File: web/filament-js/jsbindings.cpp [hunk 2]
```diff
@@ -1411,7 +1411,10 @@ class_<MaterialInstance>("MaterialInstance")
     .function("setStencilWriteMask", EMBIND_LAMBDA(void,
             (MaterialInstance* self, uint8_t writeMask), {
                 self->setStencilWriteMask(writeMask, backend::StencilFace::FRONT_AND_BACK);
-            }), allow_raw_pointers());
+            }), allow_raw_pointers())
+    .function("setFrontFaceWindingInverted", &MaterialInstance::setFrontFaceWindingInverted)
+    .function("isFrontFaceWindingInverted", &MaterialInstance::isFrontFaceWindingInverted);
```
Comment (pixelflinger) [inline web/filament-js/jsbindings.cpp:1416]: we don't need to remove the View's version? were they missing?

**Review by pixelflinger — CHANGES_REQUESTED** [review]
In this particular loop I like to be explicit about the dereference that can be hoisted outside it.

File: filament/src/RenderPass.cpp (hunk 1 above)
Comment (poweifeng) [reply filament/src/RenderPass.cpp]: done

File: web/filament-js/jsbindings.cpp (hunk 2 above)
Comment (poweifeng) [reply web/filament-js/jsbindings.cpp:1416]: yes, the View methods were missing.

**Review by pixelflinger — APPROVED** [review]

**Pull request merged**